nixos/modules/services/databases/postgresql.nix

```diff
@@ -450,11 +452,14 @@ in
 
   config = mkIf cfg.enable {
 
-    assertions = map ({ name, ensureDBOwnership, ... }: {
-      assertion = ensureDBOwnership -> builtins.elem name cfg.ensureDatabases;
+    assertions = map ({ name, ensureDBOwnership, ... }: let
+        dbOwnershipList = if builtins.isList ensureDBOwnership then ensureDBOwnership else if ensureDBOwnership then [name] else [];
```

Comment:
Rather than repeat this multiple time, you can use `apply` in the option definition to tranform a `true` into a singleton list `[ name ]`.

Reply:
I tried that but couldn't figure out how to use the value of `name` within the `apply` function

Reply:
`types.submodule` can use an argument list which would allow you to get at the name IIRC.

Currently on my phone so it's difficult to check.